internal/test/envtest/environment.go

```diff
@@ -422,6 +422,28 @@ func (e *Environment) CreateAndWait(ctx context.Context, obj client.Object, opts
 	return nil
 }
 
+// DeleteAndWait deletes the given object and waits for it to be not available from the API Server.
+func (e *Environment) DeleteAndWait(ctx context.Context, obj client.Object) error {
```

Comment:
This is very close to CleanupAndWait (takes one object instead of several) but I think it's worth adding just for the semantics of being closer to the normal client calls.

Reply:
(is it the only difference?)

Reply:
+1 to rename current func to DeleteAndWait + eventually adding 

```suggestion
// Deprecated
func (e *Environment) CleanupAndWait(ctx context.Context, objs ...client.Object) error {
    return DeleteAndWait(ctx, objs...)
}
```

Reply:
In my opinion we don't have to deprecate as it's all just internal: `[internal/test/envtest/environment.go](https://github.com/kubernetes-sigs/cluster-api/pull/7589/files#diff-1251738b05a88d7649dd5334cdede69a3658cc095972d095abea116b59d34b3c)`

Reply:
I hadn't realized this was in internal - I'll rename the func and update this PR.

Reply:
I assumed that :)

Reply:
Can't actually create a 'Delete' function here as it interferes with the Client interface (answering probably how it ended up with this name in the first place), and I don't like having `Cleanup` paired with `DeleteAndWait` I think the easiest thing to do is just to continue to use cleanup as this is really not important :smile: